src/fancy-tools/freehandMouseTool.js

```diff
@@ -0,0 +1,1080 @@
+/* eslint no-loop-func: 0 */ // --> OFF
+/* eslint no-underscore-dangle: 0 */
+import EVENTS from './../events.js';
+import external from './../externalModules.js';
+import baseAnnotationTool from './../base/baseAnnotationTool.js';
+// State
+import { addToolState, getToolState } from './../stateManagement/toolState.js';
+import toolStyle from './../stateManagement/toolStyle.js';
+import toolColors from './../stateManagement/toolColors.js';
+import { mutations } from '../store/index.js';
+// Manipulators
+import drawHandles from './../manipulators/drawHandles.js';
+import { findAndMoveHandleNearImagePoint } from '../eventDispatchers/mouseEventHandlers/mouseDown.js';
+// Implementation Logic
+import pointInsideBoundingBox from '../util/pointInsideBoundingBox.js';
+import calculateSUV from '../util/calculateSUV.js';
+import numbersWithCommas from './shared/numbersWithCommas.js';
+import insertOrDelete from './shared/freehandUtils/insertOrDelete.js';
+import freehandArea from './shared/freehandUtils/freehandArea.js';
+import calculateFreehandStatistics from './shared/freehandUtils/calculateFreehandStatistics.js';
+import freehandIntersect from './shared/freehandUtils/freehandIntersect.js';
+import { FreehandHandleData } from './shared/freehandUtils/FreehandHandleData.js';
+// Drawing
+import { getNewContext, draw, drawJoinedLines } from './../util/drawing.js';
+import drawLinkedTextBox from './../util/drawLinkedTextBox.js';
+import { clipToBox } from '../util/clip.js';
+
+export default class extends baseAnnotationTool {
+
+  constructor (name) {
+    super({
+      name: name || 'freehandMouse',
+      supportedInteractionTypes: ['mouse'],
+      configuration: defaultFreehandConfiguration()
+    });
+
+    // Create bound callback functions for private event loops
+    this._drawingMouseDownCallback = this._drawingMouseDownCallback.bind(this);
+    this._drawingMouseMoveCallback = this._drawingMouseMoveCallback.bind(this);
+
+    this._editMouseUpCallback = this._editMouseUpCallback.bind(this);
+    this._editMouseDragCallback = this._editMouseDragCallback.bind(this);
+  }
+
+  /**
+   * Create the measurement data for this tool
+   *
+   * @param {*} eventData
+   * @returns
+   */
+  createNewMeasurement (eventData) {
+
+    const goodEventData =
+      eventData && eventData.currentPoints && eventData.currentPoints.image;
+
+    if (!goodEventData) {
+      console.error(
```

Comment:
Any reason why we shouldn't just throw an error?

Reply:
Nope, that would make more sense 👍 .